addons/web_editor/static/src/js/wysiwyg/widgets/image_crop_widget.js

```diff
@@ -1,174 +1,228 @@
-            }));
+
+        // Get id, mimetype and originalSrc.
+        const {attachment, original} = await this._rpc({
+            route: '/web_editor/get_image_info',
+            params: {src: this.initialSrc.split(/[?#]/)[0]},
+        });
+        if (!attachment) {
+            // Local image that doesn't have an attachment, don't allow crop?
+            // In practice, this can happen if an image is directly linked with its
+            // static url and there is no corresponding attachment, (eg, logo in mass_mailing)
+            this.isCroppable = false;
+            return;
         }
-        return Promise.all(defs);
+        this.originalId = original.id;
+        this.originalSrc = original.image_src;
+        this.originalName = original.name;
+        this.mimetype = original.mimetype;
+        this.attachmentId = attachment.id;
     },
     /**
      * @override
      */
-    start: function () {
-        this.$cropperImage = this.$('.o_cropper_image');
-        if (this.$cropperImage.length) {
-            var data = this.$media.data();
-            var ratio = 0;
-            for (var i = 0; i < this.aspectRatioList.length; i++) {
-                if (this.aspectRatioList[i][1] === data.aspectRatio) {
-                    ratio = this.aspectRatioList[i][2];
-                    break;
-                }
-            }
-            this.$cropperImage.cropper({
-                viewMode: 2,
-                dragMode: 'move',
-                autoCropArea: 1.0,
-                aspectRatio: ratio,
-                data: _.pick(data, 'x', 'y', 'width', 'height', 'rotate', 'scaleX', 'scaleY')
+    async start() {
+        if (!this.isCroppable) {
+            this.displayNotification({
+              type: 'warning',
+              title: _t("This image is an external image"),
+              message: _t("This type of image is not supported for cropping.<br/>If you want to crop it, please first download it from the original source and upload it in Odoo."),
             });
+            return this.destroy();
         }
-        return this._super.apply(this, arguments);
+        const _super = this._super.bind(this);
+        const cropperWrapper = this.$('.o_we_cropper_wrapper')[0];
+
+        // Replacing the src with the original's so that the layout is correct.
+        this.media.setAttribute('src', this.originalSrc);
+        await new Promise(resolve => this.media.addEventListener('load', resolve, {once: true}));
+        this.$cropperImage = this.$('.o_we_cropper_img');
+        const cropperImage = this.$cropperImage[0];
+        [cropperImage.style.width, cropperImage.style.height] = [this.$media.width() + 'px', this.$media.height() + 'px'];
+
+        // Overlaying the cropper image over the real image
+        $(cropperWrapper).offset(this.$media.offset());
+        // The cropper takes the size of the container, including padding
+        // The image doesn't fill its padding, so we use margin on the container
```

Comment:
Can it? How?

Reply:
Maybe I did not understand the comment, can you explain here?

Reply:
If you use padding directly on an `<img>`, the image does not cover the padding, but the `width()` function returns the inner width of the element which includes padding. And the cropper lib uses the inner size of the container, so if we make the container the size of the `<img>`, it's actually larger than the display size of the image, and won't overlay over it correctly.

Reply:
> width() function returns the inner width of the element which includes padding

`width()` **excludes** padding, that's why you use it to size the wrapper... which is why your `// The cropper takes the size of the container, including padding` comment confuses me: your cropper is the size of the image excluding padding, like it should... :)

Reply:
I'm just confused. The thing is that the container aligns itself with the image's offset that points to the top left corner of the image padding, the margin serves to align the container with the actual image rather than its padding. I'll reword the comment to make that clearer.